readme.md

```diff
@@ -246,123 +240,157 @@ Defaults to `'Too many requests, please try again later.'`
+
+(Technically they are counted and then un-counted, so a large number of slow
+requests all at once could still trigger a rate-limit. This may be fixed in a
+future release. PRs welcome!)
+
+Defaults to `false`.
+
+### `skipSuccessfulRequests`
+
+> `boolean`
+
+If `true`, the library will (by default) skip all requests that are considered
+'failed' by the `requestWasSuccessful` function. By default, this means requests
+succeed when the response status code < 400.
+
+(Technically they are counted and then un-counted, so a large number of slow
+requests all at once could still trigger a rate-limit. This may be fixed in a
+future release. PRs welcome!)
+
+Defaults to `false`.
+
 ### `keyGenerator`
 
-Function used to generate keys.
+> `function`
 
-Defaults to `request.ip`, similar to this:
+Method to generate custom identifiers for clients.
+
+By default, the client's IP address is used:
 
 ```ts
-const keyGenerator = (request /*, response*/) => request.ip
+const keyGenerator = (request, response) => request.ip
 ```
 
 ### `handler`
 
-The function to handle requests once the max limit is exceeded. It receives the
-`request` and the `response` objects. The `next` param is available if you need
-to pass to the next middleware/route. Finally, the `options` param has all of
-the options that originally passed in when creating the current limiter and the
-default values for other options.
+> `function`
 
-The `request.rateLimit` object has `limit`, `current`, and `remaining` number of
-requests and, if the store provides it, a `resetTime` Date object.
+Express request handler that sends back a response when a client is
+rate-limited.
 
-Defaults to:
+By default, sends back the `statusCode` and `message` set via the options:
 
 ```ts
 const handler = (request, response, next, options) => {
 	response.status(options.statusCode).send(options.message)
 }
 ```
 
-### `requestWasSuccessful`
+### `onLimitReached`
 
-Function that is called when `skipFailedRequests` and/or
-`skipSuccessfulRequests` are set to `true`. May be overridden if, for example, a
-service sends out a 200 status code on errors.
+> `function`
 
-Defaults to
+Express request handler that sends back a response when a client has reached
+their rate limit, and will be rate limited on their next request.
 
-```ts
-const requestWasSuccessful = (request, response) => response.statusCode < 400
-```
+This method was
+[deprecated in v6](https://github.com/nfriedly/express-rate-limit/releases/v6.0.0) -
+Please use a custom `handler` that checks the number of hits instead.
 
-### `skipFailedRequests`
+### `skip`
 
-When set to `true`, failed requests won't be counted. Request considered failed
-when:
+> `function`
 
-- response status >= 400
-- requests that were cancelled before last chunk of data was sent (response
-  `close` event triggered)
-- response `error` event was triggered by response
+Method (in the form of middleware) to determine whether or not this request
```

Comment:
Again, I don't like calling it middleware, because it's supposed to return a value.

Also, we should mention that it may be async / return a promise

Reply:
Agree, will change this